packages/material-ui/test/integration/Menu.test.js

```diff
@@ -1,403 +1,313 @@
 
-    it('should change focus to the 2nd item when up arrow is pressed', () => {
-      TestUtils.Simulate.keyDown(portalLayer.querySelector('ul'), {
-        key: 'ArrowUp',
-      });
-      assert.strictEqual(document.activeElement, portalLayer.querySelectorAll('li')[1]);
-    });
+    fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+    expect(getAllByRole('menuitem')[0]).to.be.focused;
 
-    it('should select the 2nd item and close the menu', () => {
-      portalLayer.querySelectorAll('li')[1].click();
-      assert.strictEqual(wrapper.text(), 'selectedIndex: 1, open: false');
-    });
-  });
+    fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+    expect(getAllByRole('menuitem')[2]).to.be.focused;
 
-  describe('opening with a selected item', () => {
-    let wrapper;
+    fireEvent.keyDown(document.activeElement, { key: 'Home' });
+    expect(getAllByRole('menuitem')[0]).to.be.focused;
 
-    before(() => {
-      wrapper = mount(<SimpleMenu transitionDuration={0} selectedIndex={2} />);
-    });
+    fireEvent.keyDown(document.activeElement, { key: 'End' });
+    expect(getAllByRole('menuitem')[2]).to.be.focused;
 
-    it('should not be open', () => {
-      const popover = wrapper.find(Popover);
-      assert.strictEqual(popover.props().open, false);
-      const menuEl = document.getElementById('simple-menu');
-      assert.strictEqual(menuEl, null);
-    });
+    fireEvent.keyDown(document.activeElement, { key: 'ArrowRight' });
+    expect(getAllByRole('menuitem')[2], 'no change on unassociated keys').to.be.focused;
+  });
 
-    it('should focus the 3rd selected item', () => {
-      wrapper.find('button').simulate('click');
-      const portalLayer = document.querySelector('[data-mui-test="Modal"]');
-      assert.strictEqual(document.activeElement, portalLayer.querySelectorAll('li')[2]);
-    });
+  it('focuses the selected item when opening', () => {
+    const { getAllByRole, getByLabelText } = render(
+      <SimpleMenu transitionDuration={0} selectedIndex={2} />,
+    );
+    const button = getByLabelText('Open menu');
 
-    it('should select the 2nd item and close the menu', () => {
-      const portalLayer = document.querySelector('[data-mui-test="Modal"]');
-      const item = portalLayer.querySelector('ul').children[1];
-      item.click();
-      assert.strictEqual(wrapper.text(), 'selectedIndex: 1, open: false');
-    });
+    expect(document.body).to.be.focused;
 
-    it('should focus the 2nd selected item', () => {
-      wrapper.find('button').simulate('click');
-      const portalLayer = document.querySelector('[data-mui-test="Modal"]');
-      assert.strictEqual(document.activeElement, portalLayer.querySelectorAll('li')[1]);
-    });
+    button.focus();
+    fireEvent.click(button);
+    expect(getAllByRole('menuitem')[2]).to.be.focused;
   });
 
   describe('Menu variant differences', () => {
-    const contentAnchorTracker = [false, false, false];
```

Comment:
What was tested here?

Reply:
> What was tested here?

The difference between the `menu` and `selectedMenu` variants is that the `menu` variant ignores whether or not an item is selected. The key differences then being that for the `selectedMenu` variant, focus is placed on the selected item and the selected item is used as the content anchor, whereas for the `menu` variant, focus is always on the list (I've started writing up my explanation for this) and the content anchor is always the first valid item. The `contentAnchorTracker` is very dependent on implementation details of how Popover uses the content anchor to determine the position of the popover.

Reply:
I don't understand what a content anchor is (what does the user see?) and how it relates to offsetTop?

Reply:
The content anchor is the element within the popover that gets vertically aligned with the anchor element. So if item 2 is selected, then when the menu opens item 2 will be aligned with the menu button.

Reply:
Ah makes sense. Probably better to use a visual regression test for that.